doc/branding.md

```diff
@@ -41,13 +41,88 @@ Verify the result after a build by inspecting:
     printed on a label on the device.
 
 
-Factory Defaults
-----------------
+Factory & Failure Config
+------------------------
 
-The Infix default configuration, factory-config, is part static files
-and part per-device generated files, e.g., SSH hostkey and hostname.
-The latter is constructed from the file `/etc/hostname`, appended with
-the last three octets of the system's base MAC address.  To override the
+To support booting the same image (CPU architecture) on multiple boards,
+Infix by default generates the device's initial configuration every time
+at boot.  This also ensures the device can always be restored to a known
+state after a factory reset, since the `factory-config` is guaranteed to
+be compatible with the YANG models for the given software version. (For
+more information on how the system boots, please see the section [Key
+Concepts](introduction.md#key-concepts) in the Introduction document.)
+
+However, for custom builds of Infix it is possible to override this with
+a single static `/etc/factory-config.cfg` (and failure-config) in your
+rootfs overlay -- with a [VPD](vpd.md) you can even support several!
+
+
+### Variables & Format Specifiers
+
+Parts of the configuration you likely always want to generated, like the
+SSH hostkey used by NETCONF, a unique hostname, or the `admin` user's
+unique (per-device with a VPD) password hash.  This section lists the
+available keywords, see the next section for examples of how to use
+them:
+
+ - **Default password hash:** `$factory$` (from VPD, .dtb, or built-in)
+ - **Default NETCONF hostkey:** `genkey` (regenerated at factory reset)
+ - **Hostname format specifiers:**
+   - `%i`: OS ID, from `/etc/os-release`, from Menuconfig branding
+   - `%h`: Default hostname, from `/etc/os-release`, from branding
+   - `%m`: NIC specific part of base MAC, e.g., to `c0-ff-ee`
+   - `%%`: Literal %
```

Comment:
This is great information to have! :+1:

Should we explicitly state the XPaths where these variables can be used?

Reply:
Good point, thanks!  Will update and force-push (adding [skipci] as well).